Do not try to resolve temporary tables from global context

azat opened the pull request:
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Cc @tavplubix

tavplubix reviewed (approved):
LGTM, but let's also throw `LOGICAL_ERROR` on attempt to add temporary table into global context.

tavplubix commented on src/Interpreters/Context.cpp:
```diff
@@ -2821,6 +2821,10 @@ StorageID Context::resolveStorageIDImpl(StorageID storage_id, StorageNamespace w
 
     bool look_for_external_table = where & StorageNamespace::ResolveExternal;
+    /// Global context should not contain temporary tables
+    if (isGlobalContext() && getApplicationType() != ApplicationType::LOCAL)
```
Btw, the second part of the condition (`getApplicationType() != ApplicationType::LOCAL`) is probably outdated. AFAIK, after #26231 clickhouse-local executes queries in separate query context, not in global one.

azat replied:
Indeed, nice catch! Fixed.

azat commented:
>LGTM, but let's also throw LOGICAL_ERROR on attempt to add temporary table into global context.

There is already assert(), but OK, let's convert it to LOGICAL_ERROR.

azat committed:
Do not try to resolve temporary tables from global context

azat committed:
Remove check for clickhouse-local

From #26231 clickhouse-local uses local context for query execution.

azat committed:
Convert assert(!isGlobalContext()) to LOGICAL_ERROR

azat committed:
Do not use global context for query execution in clickhouse-copier

azat commented:
>Integration tests (asan) — fail: 19, passed: 1730, flaky: 0
>[Integration tests (release) — fail: 20, passed: 1230, flaky: 1](https://clickhouse-test-reports.s3.yandex.net/31333/2e4ef4d20aaaf958d7ab21838387c954b5089be7/integration_tests_(release).html#fail1)
>Integration tests (thread) — fail: 18, passed: 1732, flaky: 0

- `test_cluster_copier` - 652394e1b6bf9b550cdd9d702ba13c826d2e6514
- `test_replicated_fetches_bandwidth` - flaky
- `test_storage_kafka` - flaky

azat committed:
Disable log_queries in clickhouse-copier

This will avoid calling Context::addQueryFactoriesInfo()

v2: disable it in ClusterCopier::reloadTaskDescription() too
v3: just create local context and pass it to the ClusterCopier instead
of global context
